src/blueprints/tv2/action-factories/tv2-action-factory-provider.ts

```diff
@@ -0,0 +1,328 @@
+} from '../timeline-object-factories/interfaces/tv2-video-mixer-timeline-object-factory'
+import {
+  Tv2AudioTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-audio-timeline-object-factory'
+import { DeviceType } from '../../../model/enums/device-type'
+import { Tv2LoggerFacade } from '../tv2-logger-facade'
+import {
+  Tv2TriCasterVideoMixerTimelineObjectFactory
+} from '../timeline-object-factories/tv2-tri-caster-video-mixer-timeline-object-factory'
+import { AtemToTriCasterSplitScreenConverter } from '../helpers/atem-to-tricaster-split-screen-converter'
+import { Tv2MisconfigurationException } from '../exceptions/tv2-misconfiguration-exception'
+import {
+  Tv2SisyfosAudioTimelineObjectFactory
+} from '../timeline-object-factories/tv2-sisyfos-audio-timeline-object-factory'
+import {
+  Tv2VideoClipTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-video-clip-timeline-object-factory'
+import { Tv2AssetPathHelper } from '../helpers/tv2-asset-path-helper'
+import { Tv2CasparCgTimelineObjectFactory } from '../timeline-object-factories/tv2-caspar-cg-timeline-object-factory'
+import { Tv2ActionManifestMapper } from '../helpers/tv2-action-manifest-mapper'
+import {
+  Tv2GraphicsTimelineObjectFactoryFactory
+} from '../timeline-object-factories/tv2-graphics-timeline-object-factory-factory'
+import { Tv2StringHashConverter } from '../helpers/tv2-string-hash-converter'
+import {
+  Tv2GraphicsSplitScreenTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-graphics-split-screen-timeline-object-factory'
+import {
+  Tv2RobotTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-robot-timeline-object-factory'
+import {
+  Tv2TelemetricsTimelineObjectFactory
+} from '../timeline-object-factories/tv2-telemetrics-timeline-object-factory'
+
+
+interface ActionFactoryInstance<T> {
+  factory: T,
+  shouldFactoryBeRecreated: (configuration: Tv2BlueprintConfiguration) => boolean
+}
+
+export class Tv2ActionFactoryProvider {
+
+  private cameraActionFactoryInstance: ActionFactoryInstance<Tv2CameraActionFactory>
+  private remoteActionFactoryInstance: ActionFactoryInstance<Tv2RemoteActionFactory>
+  private transitionEffectActionFactoryInstance: ActionFactoryInstance<Tv2TransitionEffectActionFactory>
+  private audioActionFactoryInstance: ActionFactoryInstance<Tv2AudioActionFactory>
+  private graphicsActionFactoryInstance: ActionFactoryInstance<Tv2GraphicsActionFactory>
+  private videoClipActionFactoryInstance: ActionFactoryInstance<Tv2VideoClipActionFactory>
+  private videoMixerConfigurationActionFactoryInstance: ActionFactoryInstance<Tv2VideoMixerConfigurationActionFactory>
+  private splitScreenActionFactoryInstance: ActionFactoryInstance<Tv2SplitScreenActionFactory>
+  private replayActionFactoryInstance: ActionFactoryInstance<Tv2ReplayActionFactory>
+  private robotActionFactoryInstance: ActionFactoryInstance<Tv2RobotActionFactory>
```

Comment:
These might be undefined :)

Reply:
Only at instantiation of the class. They aren't allow to be assigned undefined.
If we should change anything we should have a constructor that initializes them

Reply:
Also after instantiation of the class. You would have to call all methods (that can cache a factory) for them to not be undefined.

I like the idea to have them initialized in a constructor.